atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraAtlasDbFactory.java

```diff
@@ -106,27 +100,32 @@ private static CassandraKeyValueServiceConfig toCassandraConfig(KeyValueServiceC
     }
 
     @VisibleForTesting
-    Supplier<CassandraKeyValueServiceRuntimeConfig> preprocessKvsRuntimeConfig(
-            Supplier<Optional<KeyValueServiceRuntimeConfig>> runtimeConfig) {
-        return () -> {
-            Optional<KeyValueServiceRuntimeConfig> configOptional = runtimeConfig.get();
-
-            return configOptional
-                    .map(config -> {
-                        if (!(config instanceof CassandraKeyValueServiceRuntimeConfig)) {
-                            log.error(
-                                    "Invalid KeyValueServiceRuntimeConfig. Expected a KeyValueServiceRuntimeConfig of"
+    Refreshable<CassandraKeyValueServiceRuntimeConfig> preprocessKvsRuntimeConfig(
+            Refreshable<Optional<KeyValueServiceRuntimeConfig>> runtimeConfig) {
+        return runtimeConfig.map(maybeConfig -> maybeConfig
+                .map(config -> {
+                    if (!(config instanceof CassandraKeyValueServiceRuntimeConfig)) {
+                        log.error(
```

Comment:
This whole thing remains a little jank, but I couldn't find an easy way to go full refreshables since this method has a default if initial map fails.

Regardless, this whole method will be massively simplified after everyone has migrated from install to runtime, since we'll no longer need the default.